.github/workflows/linting.yml

```diff
@@ -35,3 +35,18 @@ jobs:
             echo "For example: \`gersemi -i .\`"
             exit 1
           fi
+
+  source-formatting:
+    name: Source Code Formatting
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y clang-format-14
+      - name: Run clang-format
+        run: |
+          FILES=$(find . -type f ! -path '**/build/*' | egrep '\.(cc|cp|cpp|c++|cxx|cu|cuh|hh|hpp|hxx|hip|vert|frag)$')
+          clang-format -style=file -i $FILES
+          git diff --exit-code
```

Comment:
It's probably better to use the latest version. I'd suggest doing something like:
```yaml
env:
  CLANG-FORMAT: /usr/lib/llvm-18/bin/clang-format

jobs:
  ...

  source-formatting:
    name: Source Code Formatting
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Install dependencies
        run: |
          sudo apt-get update -qq
          apt-get install -y git wget
          wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | sudo tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
          sudo echo "deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main" | sudo tee -a /etc/apt/sources.list.d/llvm-18.list
          sudo echo "deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-18 main" | sudo tee -a /etc/apt/sources.list.d/llvm-18.list
          sudo apt-get update -qq
          sudo apt-get install -y clang-format-18
      - name: Run clang-format
        run: |
          FILES=$(find . -type f ! -path '**/build/*' | egrep '\.(cc|cp|cpp|c++|cxx|cu|cuh|hh|hpp|hxx|hip|vert|frag)$')
          ${{env.CLANG-FORMAT}} -style=file -i $FILES
          git diff --exit-code
```
(new formatting errors on source code are reported when using the newer version)

Reply:
Also, I'm not sure if this is the desired behaviour but currently `clang-format` is run for all files applicable (whether they were modified or not). If we want to run it only for the modified files (as the workflow is only triggered on `push` or `pull_request` on some branches) we could do:
```yaml
      - name: Run clang-format
        run: |
          git config --global --add safe.directory $GITHUB_WORKSPACE
          Scripts/CodeFormat/check_format.sh --binary "${{env.CLANG_FORMAT}}"
```
(the `check_format.sh` script already takes care of selecting the proper base commit)

Reply:
The current behavior of running `clang-format` for all applicable files is intentional, as this is a CI action meant to ensure consistent formatting across the entire codebase.

Reply:
I think scanning the entire codebase is okay for the status check. 
If we want to reuse the check_format.sh script to do that, then that may help for consistency. Also, we won't have to maintain two lists of extensions to scan.